common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -96,6 +96,10 @@ public Object run() {
                         return e;
                     } catch (IllegalAccessException e) {
                         return e;
+                    } catch (NoClassDefFoundError e) {
```

Comment:
I think LinkageError might be a little more robust here, would love to hear what @jasontedor has to say.

Reply:
Yeah maybe that would be a good idea!

Reply:
Thanks for the ping @johnou. I think the most specific error possible should be caught. If this throws a `NoClassDefFoundError` in an OSGI situation, that I think that's what should be caught here only.

Reply:
Also, before merging, I would recommend thinking about my [comment](https://github.com/netty/netty/issues/6548#issuecomment-346465502) on the original issue. It's hard to support something that you do not have experience with, tests for, etc. so maybe the best path is not to make a code change here at all (because a future refactoring might "break" it again). Instead, there's an out here for users in OSGI already: either add the dependency needed, or set a flag to skip trying to use unsafe.

Reply:
There is actually a Netty OSGI test suite 💯

Reply:
yep ... I will try to add a test.. not idea if I will be successful tho =P

Reply:
@normanmaurer not as easy as trying to load the class with the PaxExam runner is it? :cheese:

Reply:
Apparently it does not do much of anything as it's not catching an issue for OSGI environments that do not add the necessary dependency for unsafe that has been there for over a year (#5640). And I question having that with two of the main maintainers saying that they do not have OSGI knowledge to maintain support for it. And none of this negates my point that there's an out for OSGI already without a change needed in Netty.

Reply:
@jasontedor I would still argue that this specific exception is not 100% related to OSGI and that Netty should be able to detect if unsafe is available or not and if it is not, fallback.

Reply:
Okay, I agree with you.